widget/form_test.go

```diff
@@ -46,9 +47,53 @@ func TestForm_CustomButtonsText(t *testing.T) {
 	assert.Equal(t, "Submit", form.SubmitText)
 	assert.Equal(t, "Cancel", form.CancelText)
 
-	form := NewForm().
+	form = NewForm().
 		OnSubmitWithLabel("Apply", func() {}).
 		OnCancelWithLabel("Close", func() {})
 	assert.Equal(t, "Apply", form.SubmitText)
 	assert.Equal(t, "Close", form.CancelText)
 }
+
+func TestForm_AddRemoveButton(t *testing.T) {
+	scount := 0
+	ccount := 0
+	sscount := 10
+	tapped := make(chan bool)
+	form := NewForm().
+		Append("test", NewEntry()).
+		OnSubmit(func() { scount++; tapped <- true }).
+		OnCancel(func() { ccount++; tapped <- true })
+
+	go test.Tap(form.submitButton)
+	func() {
+		select {
+		case <-tapped:
+		case <-time.After(1 * time.Second):
+			assert.Fail(t, "Timed out waiting for submit button tap")
+		}
+	}()
+	assert.Equal(t, 1, scount)
```

Comment:
Why do you run the tap in a goroutine? I think the whole async part can be removed:
```go
test.Tap(…)
assert.Equal(t, 1, scount)
```

Reply:
copypasta from the button_test code

I think its there to make sure that the if the button press never happens, it doesnt hang the test ?

But yeah, that looks much simpler, will do

Reply:
that works :+1: